src/pudl/transform/phmsagas.py

```diff
@@ -0,0 +1,199 @@
+
+
+@asset(
+    ins={"raw_data": AssetIn(key="raw_phmsagas__yearly_distribution")},
+    io_manager_key=None,  # TODO: check on this one ("pudl_io_manager"? something else?)
+    compute_kind=None,  # TODO: check on this one
+)
+def core_phmsagas__yearly_distribution_operators(
+    raw_data: pd.DataFrame,
+) -> pd.DataFrame:
+    """Build the :ref:`core_phmsagas__yearly_distribution_operators`."""
+    df = raw_data.loc[
+        :, YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["columns_to_keep"]
+    ].copy()
+
+    # Specify the columns to convert to integer type
+    cols_to_convert = YEARLY_DISTRIBUTION_OPERATORS_COLUMNS[
+        "columns_to_convert_to_ints"
+    ]
+
+    # Fill NaN values with pd.NA, then cast to "Int64" nullable integer type
+    df[cols_to_convert] = (
+        df[cols_to_convert]
+        .apply(lambda col: col.where(col.notna(), pd.NA))
+        .astype("Int64")
+    )
+
+    # Ensure all "report_year" values have four digits
+    mask = df["report_year"] < 100
+
+    # Convert 2-digit years to appropriate 4-digit format (assume cutoff at year 50)
+    # We could also use the first 4 digits of the "report_number" but there was at least one anomaly here with an invalid year
+    df.loc[mask, "report_year"] = df.loc[mask, "report_year"].apply(
+        lambda x: 2000 + x if x < 50 else 1900 + x
+    )
+
+    # Fill in bad strings
+    df = standardize_na_values(df)
+
+    # Standardize case for city, county, operator name, etc.
+    # Capitalize the first letter of each word in all object-type columns except the excluded ones
+    df[
+        df.select_dtypes(include=["object"]).columns.difference(
+            YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["capitalization_exclusion"]
+        )
+    ] = df[
+        df.select_dtypes(include=["object"]).columns.difference(
+            YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["capitalization_exclusion"]
+        )
+    ].apply(lambda col: col.str.title())
+
+    # Standardize state abbreviations
+    # First create a dictionary of state names to abbreviations
+    state_to_abbr = {
+        x.subdivision_name: x.subdivision_code
+        for x in POLITICAL_SUBDIVISIONS.itertuples()
+        if x.country_code == "USA" and x.subdivision_type == "state"
+    }
+    # Add abbreviations to the dictionary so we can make sure the abbreviations in the raw data exist
+    state_to_abbr.update(
+        {
+            x.subdivision_code: x.subdivision_code
+            for x in POLITICAL_SUBDIVISIONS.itertuples()
+            if x.country_code == "USA" and x.subdivision_type == "state"
+        }
+    )
+
+    # Function for looking up state names and abbreviations
+    def standardize_state(state):
+        if pd.isna(state):
+            return state
+        state = state.strip()
+        standardized_state = state_to_abbr.get(state, state)
+        if standardized_state not in state_to_abbr.values():
+            return np.nan
+        return standardized_state
```

Comment:
Also note that you need to be careful to avoid mixing incompatible NA types.  `np.nan` is only appropriate for numpy floating point values. If it gets mixed in with other types in a column, the whole column has to be treated as an `object`.

For all the nullable Pandas dtypes (e.g. `Int64` and `string`) the appropriate NA value will be `pd.NA`.

Reply:
Ah whoops thanks for catching that. Fixing that now.

Reply:
Updated